File: 백준/Silver/21921. 블로그/블로그.py
import sys

N, X = map(int, sys.stdin.readline().split())
visited = list(map(int, sys.stdin.readline().strip().split()))

window_sum = sum(visited[:X])
result = [window_sum]

for i in range(X, N):
    window_sum += visited[i] - visited[i-X]
    result.append(window_sum)

max_visited = max(result)

if max_visited > 0:
    count = 0
    for i in result:
        if i == max_visited:
            count += 1
    print(max_visited)
    print(count)

else:
    print("SAD")

File: 백준/Silver/21921. 블로그/블로그.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class Main {
	
	static int N, X;
	static ArrayList<Integer> visited, sumArr;
	
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		N = Integer.parseInt(st.nextToken());
		X = Integer.parseInt(st.nextToken());
				
		visited = new ArrayList<>();
		st = new StringTokenizer(br.readLine());
		for(int i=0; i<N; i++) {
			visited.add(Integer.parseInt(st.nextToken()));
		}
		
		// 처음 윈도우 계산
		int windowSum = 0;
		for(int i=0; i<X; i++) {
			windowSum += visited.get(i);
		}
		sumArr = new ArrayList<>();
		sumArr.add(windowSum);
		
		
		// 윈도우 시작
		int max_visited = windowSum;
		for(int i=X; i<N; i++) {
			windowSum += visited.get(i) - visited.get(i-X);
			sumArr.add(windowSum);
			max_visited = Math.max(max_visited, windowSum);
			
		}
		
		if(max_visited > 0) {
			System.out.println(max_visited);
			int count = 0;
			for(int i=0; i<sumArr.size(); i++) {
				if(sumArr.get(i) == max_visited) {
					count += 1;
				}
			}
			System.out.println(count);
		}else {
			System.out.println("SAD");
		}

		
	}
}
